File: src/main/java/me/sd_master92/customvoting/commands/DeleteTopCommand.java
package me.sd_master92.customvoting.commands;

import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.constants.Messages;
import me.sd_master92.customvoting.subjects.VoteTopStand;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class DeleteTopCommand implements CommandExecutor
{
    private final Main plugin;

    public DeleteTopCommand(Main plugin)
    {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args)
    {
        if (sender instanceof Player)
        {
            Player player = (Player) sender;
            if (command.getPermission() != null && sender.hasPermission(command.getPermission()))
            {
                if (args.length > 0)
                {
                    try
                    {
                        int top = Integer.parseInt(args[0]);
                        VoteTopStand voteTopStand = VoteTopStand.get(top);
                        if(voteTopStand != null)
                        {
                            voteTopStand.delete(player);
                        } else
                        {
                            player.sendMessage(ChatColor.RED + "That Vote Stand does not exist.");
                        }
                    } catch (Exception e)
                    {
                        sender.sendMessage(ChatColor.RED + "Invalid argument: 'top' must be a number.");
                    }
                } else
                {
                    sender.sendMessage(ChatColor.RED + "- /deletetop <top>");
                }
            } else
            {
                player.sendMessage(Messages.NO_PERMISSION.getMessage(plugin));
            }
        } else
        {
            sender.sendMessage(Messages.MUST_BE_PLAYER.getMessage(plugin));
        }
        return true;
    }
}

File: src/main/kotlin/me/sd_master92/customvoting/commands/DeleteTopCommand.kt
package me.sd_master92.customvoting.commands

import me.sd_master92.core.command.SimpleCommand
import me.sd_master92.core.errorLog
import me.sd_master92.customvoting.CV
import me.sd_master92.customvoting.constants.enumerations.Message
import me.sd_master92.customvoting.constants.enumerations.PMessage
import me.sd_master92.customvoting.subjects.stands.VoteTopStand
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class DeleteTopCommand(private val plugin: CV) : SimpleCommand(plugin, "deletetop", false)
{
    override fun onCommand(sender: CommandSender, args: Array<String>)
    {
    }

    override fun onCommand(player: Player, args: Array<String>)
    {
        try
        {
            val top = args[0].toInt()
            val stand = VoteTopStand[plugin, top]
            if (stand.exists())
            {
                stand.delete(player)
            } else
            {
                player.sendMessage(PMessage.GENERAL_ERROR_NOT_EXIST_X.with(PMessage.VOTE_TOP_UNIT_STAND.toString()))
            }
        } catch (e: NumberFormatException)
        {
            player.sendMessage(PMessage.GENERAL_ERROR_INVALID_ARGUMENT_NOT_NUMBER_X.with("top"))
        } catch (e: Exception)
        {
            player.sendMessage(PMessage.GENERAL_ERROR.toString())
            plugin.errorLog(PMessage.VOTE_TOP_ERROR_DELETE.toString(), e)
        }
    }

    init
    {
        withPlayer(Message.MUST_BE_PLAYER.getMessage(plugin))
        withUsage(PMessage.VOTE_TOP_MESSAGE_DELETE_COMMAND_USAGE.toString())
        withNoPermMessage(Message.NO_PERMISSION.getMessage(plugin))
    }
}